sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/ParserUtils.scala

```diff
@@ -172,6 +172,20 @@ object ParserUtils {
           }
           sb.append(code.asInstanceOf[Char])
           i += 5
+        } else if ((i + 10 < strLength) && b.charAt(i + 1) == 'U' &&
```

Comment:
We could use a regex here too; might be faster. But if this isn't a hot path, probably won't matter.

Reply:
Sounds better. Thanks.

Reply:
If we are going to use`regex` to match `\U00000000` pattern, I think `regex` should also be used for other escape characters including `\u0000` and `\000`.
So I focus on 32-bit unicode escape in this PR and will refactor the code for all the escape characters with `regex`.